drools-ansible-rulebook-integration-protoextractor/src/main/java/org/drools/ansible/rulebook/integration/protoextractor/prototype/ValueCollectVisitor.java

```diff
@@ -0,0 +1,133 @@
+package org.drools.ansible.rulebook.integration.protoextractor.prototype;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ASTNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.DefaultedVisitor;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.ExtractorNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.IdentifierNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.IndexAccessorNode;
+import org.drools.ansible.rulebook.integration.protoextractor.ast.SquaredAccessorNode;
+import org.kie.api.prototype.Prototype;
+
+public class ValueCollectVisitor extends DefaultedVisitor<Object> {
```

Comment:
I introduced `ValueCollectVisitor` and simply replaced with `ValueExtractionVisitor`. But probably we will need to switch them depending on operator used in a rule condition.

Reply:
If I understand correctly the `ValueCollectVisitor` is a features superset of the `ValueExtractionVisitor` and as such probably we don't need the second anymore. Also I believe it will be hard if not impossible to discover when the second is enough or it is necessary to use the first. If all I wrote is correct maybe we can totally drop the `ValueExtractionVisitor`, but let's keep it for now.